scripts/west_commands/runners/nrf_common.py

```diff
@@ -425,22 +431,59 @@ def reset_target(self):
     def do_require(self):
         ''' Ensure the tool is installed '''
 
+    def _get_suit_starter(self):
+        suit_starter = Path(os.environ.get('ZEPHYR_BASE')) / SUIT_STARTER_PATH
+        if not suit_starter.exists():
+            raise RuntimeError("Unable to find suit manifest starter file, "
+                               "please make sure to run \'west blobs fetch "
+                               "hal_nordic\'")
+        return suit_starter.resolve()
+
+    def _check_suit_starter(self, op):
+        op = op['operation']
+        if op['type'] not in ('erase', 'recover', 'program'):
+            return None
+        elif op['type']  == 'program' and op['chip_erase_mode'] != "ERASE_UICR":
+            return None
+
+        file = self._get_suit_starter()
+        self.logger.debug(f'suit starter: {file}')
+
+        return str(file)
+
     def op_program(self, hex_file, erase, qspi_erase, defer=False, core=None):
+        args = self._op_program(hex_file, erase, qspi_erase)
+        self.exec_op('program', defer, core, **args)
+
+    def _op_program(self, hex_file, erase, qspi_erase):
         args = {'firmware': {'file': hex_file},
                 'chip_erase_mode': erase, 'verify': 'VERIFY_READ'}
         if qspi_erase:
             args['qspi_erase_mode'] = qspi_erase
-        self.exec_op('program', defer, core, **args)
+
+        return args
 
     def exec_op(self, op, defer=False, core=None, **kwargs):
-        _op = f'{op}'
-        op = {'operation': {'type': _op}}
-        if core:
-            op['core'] = core
-        op['operation'].update(kwargs)
-        self.logger.debug(f'defer: {defer} op: {op}')
-        if defer or not self.do_exec_op(op, force=False):
-            self.ops.append(op)
+
+        def _exec_op(op, defer=False, core=None, **kwargs):
+            _op = f'{op}'
+            op = {'operation': {'type': _op}}
+            if core:
+                op['core'] = core
+            op['operation'].update(kwargs)
+            self.logger.debug(f'defer: {defer} op: {op}')
+            if defer or not self.do_exec_op(op, force=False):
+                self.ops.append(op)
+            return op
+
+        _op = _exec_op(op, defer, core, **kwargs)
+        # Check if the suit manifest starter needs programming
+        if self.suit_starter: #and self.family == 'NRF54H_FAMILY':
```

Comment:
is this `and` logic should be commented? it doesn't look like a comment

Reply:
ooops. Thanks!